### PR TITLE
Enhance MIG Support Detection for NVIDIA GPUs introduced in R580

### PR DESCRIPTION
**Pull request by RuixiangMa**
## Summary
This PR significantly improves the Multi-Instance GPU (MIG) capability detection logic in the NVIDIA GPU Operator by expanding the list of supported GPU architectures and implementing a more comprehensive pattern-matching approach.

### 1. Enhanced MIG Detection Logic (`controllers/state_manager.go`)
- **Refactored** the `hasMIGCapableGPU` function to use a dedicated helper function `isMIGCapableGPUProduct`
- **Expanded** MIG support from 3 basic models to comprehensive architecture coverage
- **Implemented** structured pattern matching with clear architectural categorization

### 2. Comprehensive GPU Architecture Support
The updated detection now supports:

#### Hopper Architecture (Data Center)
- H100, H800, H200, H20，GH200 

#### Ampere Architecture  
- A100, A800, A30

#### Blackwell Architecture (Next Generation)
- GB200, B200, GB300, B300

#### Professional Workstation GPUs
- RTX PRO 6000
- RTX PRO 5000
- **Dual format support**: Both "rtx-pro-6000" and "rtx pro 6000" naming conventions

### Test Coverage
  - All supported GPU models across architectures
  - Multiple naming format variations
  - Negative test cases for non-MIG GPUs (T4, V100)
  - Edge cases (empty strings, partial matches)

### Test Results

截屏2025-11-17 16 47 44

### CODE REVIEW

File: controllers/state_manager.go [hunk 1]
```diff
@@ -289,9 +289,35 @@ func hasMIGCapableGPU(labels map[string]string) bool {
 
 	// check product label if mig.capable label does not exist
 	if value, exists := labels[gpuProductLabelKey]; exists {
-		if strings.Contains(strings.ToLower(value), "h100") ||
-			strings.Contains(strings.ToLower(value), "a100") ||
-			strings.Contains(strings.ToLower(value), "a30") {
+		return isMIGCapableGPUProduct(value)
```
Comment (rajathagasthya) [inline controllers/state_manager.go:292]: This is only a fallback check if `mig.capable` GFD label does not exist. We expect GFD to work correctly as we rely on it heavily. Given this hasn't been updated in a long time and is missing many GPUs, I don't think this change is necessary. 

cc @cdesiniotis @tariq1890 for thoughts.

Comment (tariq1890) [reply controllers/state_manager.go:292]: Agreed. I don't think this change is necessary. We don't want to maintain this static list and we want users to use GFD as the source of truth here

**Comment by rajathagasthya** [comment]
Closing as GFD provides information about MIG capability for all GPUs.
